pro/src/data_access/ApiRecipeDataAccessObject.java

```diff
@@ -1,68 +1,83 @@
 package data_access;
 
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
 import okhttp3.OkHttpClient;
 import okhttp3.Request;
 import okhttp3.Response;
 import org.json.JSONObject;
+import service.search.use_case.SearchDataAccessInterface;
 
-import java.io.IOException;
-import java.util.HashMap;
-
-
-public class ApiRecipeDataAccessObject {
+public class ApiRecipeDataAccessObject implements SearchDataAccessInterface {
     private final String keyword;
 
-    public ApiRecipeDataAccessObject(String keyword){
+    public ApiRecipeDataAccessObject(String keyword) {
         this.keyword = keyword;
```

Comment:
This attribute should not be included because when we construct a DAO in main, we have not started searching, we don't know what the keyword is, instead, we can add an attribute for recipeFactory

Reply:
Do we need a constructor in this file? The in memory user DAO in CACoding does not have a constructor, so I am not sure whether we definitely need a constructor.

Reply:
We don't need keyword but we will need recipeFactory. It follows the adapter design pattern.